lib/index.js

```diff
@@ -997,7 +997,10 @@ Mongoose.prototype.isValidObjectId = function(v) {
     v = v.toString();
   }
 
-  if (typeof v === 'string' && (v.length === 12 || v.length === 24)) {
+  if (typeof v === 'string' && v.length === 12) {
+    return true;
+  }
+  if (typeof v === 'string' && v.length === 24 && /^[a-f0-9]*$/.test(v)) {
```

Comment:
You also need to add a similar check on line 992, there's another place where we check `length === 24`